OsmoSystems/osmo-jupyter
# Get experiment data files from Google Drive format

[asana](https://app.asana.com/0/819671808102776/1147515614239923/f)

We talked about leaving this in a notebook for this ticket but it seemed obvious enough where it should go in a repo.


## Manual testing

Used on all the Drive directories in the log and eyeballed the results. see my fork of the Parse and combine calibration and sensor data snippet: http://marvin:8889/notebooks/Tools/Jupyter%20Snippets/Parse%20and%20combine%20calibration%20and%20sensor%20data-Temp%20version%20updated%20with%20Jason%20code.ipynb

this is what it took to tack the files on to a parsed data collection log:
```
from osmo_jupyter.dataset.files import get_experiment_data_files_by_type

DRIVE_EXPERIMENTS_ROOT = "/data/Google Drive/Technical/Experiments"
drive_directory_paths = parsed_log["drive_directory"].apply(
    lambda directory_name: os.path.join(DRIVE_EXPERIMENTS_ROOT, directory_name)
)
parsed_log_with_file_paths = pd.concat(
    [
        parsed_log, # parsed using `parse_data_collection_log`
        drive_directory_paths.apply(get_experiment_data_files_by_type).add_suffix(
            "_files"
        ),
    ],
    axis="columns",
)
```
![image](https://user-images.githubusercontent.com/3341011/68236255-af80d780-ffb9-11e9-96f3-a59623dbd11b.png)


*suggestion: I should probably go ahead and librarify this as well*, probably as `osmo_jupyter.dataset.parse.parse_data_collection_log_with_drive_files(log_filepath, drive_experiments_root)` but I'm not sure how much to pull on this scope creep thread before I'm stepping on Evan's ticket's toes! The logical next step after that would be to modify/provide a version of `open_and_combine_and_filter_source_data()` that's integrated with this also.
